wizard/src/components/upload_section.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Papa from "papaparse";
+import React, { useState } from "react";
+
+import { Column, CsvData, Mapping } from "../types";
+import { getPredictions } from "../utils/heuristics";
+
+const NUM_FIRST_ROWS = 3;
+const NUM_LAST_ROWS = 3;
+interface UploadSectionProps {
+  onCsvProcessed: (csv: CsvData) => void;
+  onPredictionRetrieved: (predictedMapping: Mapping) => void;
+}
+
+export function UploadSection(props: UploadSectionProps): JSX.Element {
+  const [isProcessingData, setIsProcessingData] = useState<boolean>(false);
+
+  return (
+    <div className="section-container">
+      <div className="upload-section-header">
+        <h2>Upload CSV</h2>
+      </div>
+      <div className="upload-section-inputs">
+        <input
+          type="file"
+          accept=".csv"
+          onChange={(event) => onFileUpload(event.target.files)}
+        />
+      </div>
+    </div>
+  );
+
+  function getOrderedColumns(headerRow: string[]): Column[] {
+    const orderedColumns = [];
+    headerRow.forEach((header, idx) => {
+      const colId = `${header}_${idx}`;
+      orderedColumns.push({
+        columnIdx: idx,
+        header,
+        id: colId,
+      });
+    });
+    return orderedColumns;
+  }
+
+  function onParseComplete(
+    csvData: CsvData,
+    sampleColumnValues: Map<string, Set<string>>
+  ): void {
+    for (const colId of Array.from(sampleColumnValues.keys())) {
+      csvData.columnValuesSampled.set(
+        colId,
+        Array.from(sampleColumnValues.get(colId))
+      );
+    }
+    const predictedMapping = getPredictions(csvData);
+    props.onCsvProcessed(csvData);
+    props.onPredictionRetrieved(predictedMapping);
+    setIsProcessingData(false);
+  }
+
+  function onFileUpload(files: FileList): void {
+    if (files.length < 1) {
```

Comment:
nit: how do we restrict to not more than 1?

Reply:
the input html element already restricts to only 1 file, but since it passes the files as a FileList, I thought it would be better to make the function still check that there is only 1 file, especially in case we change the html element being used